### DISCUSSION TITLE
Add `recurringCycleLimit` to discount functions

### DISCUSSION THREAD

**Discussion by gil--** [Shopify/function-examples#471] [Feedback: Discounts APIs]
Allow discounts created by functions to set the `recurringCycleLimit` so that we can dictate whether a shipping discount (or any discount) is applicable on recurring order placements.

**Comment by bkspace** [comment]
Yes! I'd just like to add that this should be available on manual and also automatic discounts (as per native Shopify discounts)

**Comment by limoniapps** [comment]
+1

**Comment by thosakwe** [comment]
Looks like Shopify has now added the necessary capabilities to their API to make this work:

https://shopify.dev/changelog/added-new-field-recurring_cycle_limit-and-applies_on_subscription-to-discountautomaticapp-and-discountautomaticappinput

**Reply by castles77** [reply]
They did.. but I'm wondering if anyone else is has found it to be an inadequate solution?

It requires you specify the `recurringCycleLimit` value when creating the AutomatcAppDiscount. So if you want to do multiple discounts for different recurrence limits you have to create an AutomatcAppDiscount for each, but you quickly run into the limit of 5 AutomatcAppDiscount.

Am I missing something here? I feel like the limit needs to be higher, or the we need to be able to return the value for recurringCycleLimit in the FunctionRunResult.

**Comment by gil--** [comment]
Closing as this is now possible: https://shopify.dev/changelog/added-new-field-recurring_cycle_limit-and-applies_on_subscription-to-discountautomaticapp-and-discountautomaticappinput
